api/v1alpha1/olsconfig_types.go

```diff
@@ -98,24 +98,63 @@ type OLSSpec struct {
 	// +kubebuilder:validation:Optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="RAG Databases",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:advanced"}
 	RAG []RAGSpec `json:"rag,omitempty"`
+	// LLM Token Quota Configuration
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="LLM Token Quota Configuration"
+	QuotaHandlersConfig QuotaHandlersConfig `json:"quotaHandlersConfig,omitempty"`
 }
 
 // RAGSpec defines how to retrieve a RAG databases.
 type RAGSpec struct {
+	// The path to the RAG database inside of the container image
 	// +kubebuilder:validation:Required
 	// +required
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Index Path in the Image"
 	IndexPath string `json:"indexPath,omitempty"`
+	// The Index ID of the RAG database
 	// +kubebuilder:validation:Required
 	// +required
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Index ID"
 	IndexID string `json:"indexID,omitempty"`
+	// The URL of the container image to use as a RAG source
 	// +kubebuilder:validation:Required
 	// +required
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Image"
 	Image string `json:"image,omitempty"`
 }
 
+// QuotaHandlersConfig defines the token quota configuration
+type QuotaHandlersConfig struct {
+	// Token quota limiters
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Token Quota Limiters",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:advanced"}
+	LimitersConfig []LimiterConfig `json:"limitersConfig,omitempty"`
+	// Enable token history
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Enable Token History",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:booleanSwitch"}
+	EnableTokenHistory bool `json:"enableTokenHistory,omitempty"`
+}
+
+// LimiterConfig defines settings for a token quota limiter
+type LimiterConfig struct {
```

Comment:
@tisnik Does input validation here (+kubebuilder:validation) make sense?

Reply:
QuotaIncrease can be zero, that way it will be reset to init at beginning of each period